broadcast/src/linked/signer/actor.rs

```diff
@@ -337,26 +366,32 @@ impl<
                     };
                     if let Err(err) = self.handle_ack(&ack).await {
                         warn!(?err, ?ack, "ack handle failed");
+                        guard.set(Status::Failure);
                         continue;
                     }
+                    guard.set(Status::Success);
                 },
 
                 // Handle completed verification futures.
-                maybe_verified = self.pending_verifies.next_completed() => {
-                    let (context, digest, verify_result) = maybe_verified;
-                    match verify_result {
+                verify = self.pending_verifies.next_completed() => {
+                    let Verify { start, context, payload, result } = verify;
+                    self.metrics.verify_duration.observe_between(start, self.context.current());
```

Comment:
Given you pass `self.context` into `self.metrics.init()`, seems "odd" to require to pass it again for any function (I wonder if it makes sense to wrap the metric to do this instead)

Reply:
Take a look at what I tried and removed here https://github.com/commonwarexyz/monorepo/pull/536/commits/f28bb0c48e0a10808216a1901dc6cd012fe25e9f

Basically since the thing that implements the `Clock` trait is usually the entire "context" there are ownership/borrowing issues when you do things like pass the reference to the context into the guard.

The other thing you can do is clone the context but that felt heavyweight to me... what do you think?

Reply:
I think the only way to do this cleanly is with `clone` (if you use `Histogram` directly).

I was thinking, instead, that the context should be stored in metrics and you then just wrap `observe_between` (calling `current` when creating the histogram guard and again when calling `observe`).